Fix import for renamed module jsonschema_spec -> jsonschema_path

This change is already merged upstream in master (n/a in a release)

--- a/openapi_core/spec/paths.py
+++ b/openapi_core/spec/paths.py
@@ -6,7 +6,7 @@
 from typing import TypeVar
 
 from jsonschema.validators import _UNSET
-from jsonschema_spec import SchemaPath
+from jsonschema_path import SchemaPath
 from openapi_spec_validator import validate
 
 TSpec = TypeVar("TSpec", bound="Spec")
